internal/rpcserver/interface.go

```diff
@@ -125,6 +125,10 @@ type ConnManager interface {
 	// the passed transactions to all connected peers.
 	RelayTransactions(txns []*dcrutil.Tx)
 
+	// AnnounceNewTransactions generates and relays inventory vectors and notifies
```

Comment:
This seems a bit cyclic in nature.  The server does indeed need to be notified of new transactions submitted via the RPC server so it can broadcast them and setup rebroadcast handling and websocket clients have to notified about the transactions as well.

Likewise, whenever new transactions arrive via the wire protocol the server needs also relay them to other peers and notify websocket clients with them.

However, this seems odd that a tx comes into the RPC server, then round trips through server before notifying Websocket clients.

This is why there was a `PeerNotifier` interface to try and make it clear that the intention is to provide the RPC with a means to notifier other peers.

This is probably something for a separate issue and not to be solved here, but I did want to note that it feels a little off to me.

Reply:
Good point.  I simplified this by just calling `s.cfg.ConnMgr.RelayTransactions` and then `s.NotifyNewTransactions` directly from the rpcserver handler for now.  There is probably more we can do here but at least this avoids the round trip from rpcserver -> server -> rpcserver that you mentioned and doesn't couple the peer notifications and ws notifications from the perspective of rpcserver.